Chisel/CHLFunctions.m

```diff
@@ -0,0 +1,65 @@
+//
+//  CHLFunctions.m
+//  Chisel
+//
+//  Copyright © 2016 Facebook. All rights reserved.
+//
+
+#import "CHLFunctions.h"
+
+// See vm_range_recorder_t
+static void enumerateRanges(task_t, void *, unsigned, vm_range_t *, unsigned);
+
+void CHLEnumerateAllocationsWithBlock(void (^block)(vm_range_t))
+{
+  /*
+   * === Overview ===
+   *
+   * To enumerate the set of live memory allocations ("ranges"), the pseudocode might look like:
+   *
+   *     for zone in all_zones()
+   *       for range in zone->ranges
+   *         callback(range)
+   *
+   * Unfortunately, the memory allocations within a zone are not exposed, so the above inner loop is not possible.
+   * Instead malloc zones provide a function that will loop over the allocations and call the given callback function.
+   */
+
+  /*
+   * === malloc API Notes ===
+   *
+   * Both `malloc_get_all_zones` and `vm_range_recorder_t` callback functions take a `task_t` and a `memory_reader_t`.
+   * However it turns out both of these can be NULL. This information isn't documented in malloc.h but can be seen in
+   * the libmalloc source.
+   *
+   * `memory_reader_t`:
+   * When NULL, the default memory reader is used which handles in process memory, and does not use the given `task_t`.
+   * `task_t`:
+   * Only required if used in a custom `memory_reader_t`, or used in a `vm_range_recorder_t` callback.
+   */
```

Comment:
I understand your "it's describing an external API" philosophical argument. My argument is far more practical: move the comments into a docblock and chisel developers can option-click to see them anytime. If you leave them where they are, they can't.

Some context: especially for relatively small projects, I prefer to make it easy for people to read documentation, even if it makes it harder to maintain that documentation. For example, rather than `see the documentation for -[MyClass myMethod] for details`, I prefer to provide a full explanation for each docblock. Similarly, here I think developers could find this information more easily if it were in a docblock, even if notes on `malloc` don't *really* belong in a docblock for a different function.

That's my personal preference, though--hence the "nit" on my original comment. :wink: 

> I just tried re-declaring `malloc_get_all_zones` inside this file, and giving that declaration a docblock. 

This is definitely overkill in my book.